runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/logging/GrpcLoggingServiceTest.java

```diff
@@ -108,9 +113,16 @@ public void testMultipleClientsFailingIsHandledGracefullyByServer() throws Excep
         tasks.add(
             () -> {
               CountDownLatch waitForTermination = new CountDownLatch(1);
-              ManagedChannel channel =
-                  InProcessChannelBuilder.forName(server.getApiServiceDescriptor().getUrl())
-                      .build();
+              String url = server.getApiServiceDescriptor().getUrl();
+              System.out.println(
+                  System.currentTimeMillis()
+                      + ": i:"
+                      + instructionId
+                      + ", thread: "
+                      + Thread.currentThread().getName()
+                      + ", url: "
+                      + url);
```

Comment:
Got the following output for the timeout. The countdown in grpc-default-executor-8 and grpc-default-executor-9 did not happen.

```
Standard Output
1619214227946: i:1, thread: pool-123-thread-1, url: InProcessServer_96
1619214227962: i:2, thread: pool-123-thread-2, url: InProcessServer_96
1619214227965: Thread: grpc-default-executor-0, about to count down: java.util.concurrent.CountDownLatch@56c8e14a[Count = 1]
1619214227971: i:1, thread: pool-123-thread-1, waiting for latch java.util.concurrent.CountDownLatch@4e14f763[Count = 1]
1619214227971: Thread: grpc-default-executor-0, finished count down: java.util.concurrent.CountDownLatch@56c8e14a[Count = 0]
1619214227965: i:2, thread: pool-123-thread-2, waiting for latch java.util.concurrent.CountDownLatch@56c8e14a[Count = 1]
1619214227972: i:3, thread: pool-123-thread-3, url: InProcessServer_96
1619214227972: i:2, thread: pool-123-thread-2, the latch(java.util.concurrent.CountDownLatch@56c8e14a[Count = 0]) returned?: true
1619214227976: i:3, thread: pool-123-thread-3, waiting for latch java.util.concurrent.CountDownLatch@16506179[Count = 1]
1619214287972: i:1, thread: pool-123-thread-1, the latch(java.util.concurrent.CountDownLatch@4e14f763[Count = 1]) returned?: false
1619214287976: i:3, thread: pool-123-thread-3, the latch(java.util.concurrent.CountDownLatch@16506179[Count = 1]) returned?: false
1619214288094: Thread: grpc-default-executor-8, about to count down: java.util.concurrent.CountDownLatch@24793c9d[Count = 1]
1619214288094: Thread: grpc-default-executor-8, finished count down: java.util.concurrent.CountDownLatch@24793c9d[Count = 0]
1619214288104: Thread: grpc-default-executor-8, about to count down: java.util.concurrent.CountDownLatch@2abec662[Count = 1]
1619214288104: Thread: grpc-default-executor-8, finished count down: java.util.concurrent.CountDownLatch@2abec662[Count = 0]
1619214288113: Thread: grpc-default-executor-9, about to count down: java.util.concurrent.CountDownLatch@3157d068[Count = 1]
1619214288114: Thread: grpc-default-executor-9, finished count down: java.util.concurrent.CountDownLatch@3157d068[Count = 0]
1619214288172: Thread: grpc-default-executor-7, about to count down: java.util.concurrent.CountDownLatch@6f4c3384[Count = 1]
1619214288172: Thread: grpc-default-executor-7, finished count down: java.util.concurrent.CountDownLatch@6f4c3384[Count = 0]
1619214288180: Thread: grpc-default-executor-8, about to count down: java.util.concurrent.CountDownLatch@6b83cc59[Count = 1]
1619214288186: Thread: grpc-default-executor-8, finished count down: java.util.concurrent.CountDownLatch@6b83cc59[Count = 0]
1619214288186: Thread: grpc-default-executor-10, about to count down: java.util.concurrent.CountDownLatch@44c364e8[Count = 1]
1619214288186: Thread: grpc-default-executor-10, finished count down: java.util.concurrent.CountDownLatch@44c364e8[Count = 0]
Standard Error
[grpc-default-executor-4] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Beam Fn Logging client connected.
[grpc-default-executor-4] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Beam Fn Logging client connected.
[grpc-default-executor-4] WARN org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Logging client failed unexpectedly.
org.apache.beam.vendor.grpc.v1p36p0.io.grpc.StatusRuntimeException: CANCELLED: client cancelled
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.Status.asRuntimeException(Status.java:526)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.stub.ServerCalls$StreamingServerCallHandler$StreamingServerCallListener.onCancel(ServerCalls.java:284)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.PartialForwardingServerCallListener.onCancel(PartialForwardingServerCallListener.java:40)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.ForwardingServerCallListener.onCancel(ForwardingServerCallListener.java:23)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onCancel(ForwardingServerCallListener.java:40)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.Contexts$ContextualizedServerCallListener.onCancel(Contexts.java:96)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closedInternal(ServerCallImpl.java:353)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closed(ServerCallImpl.java:341)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1Closed.runInContext(ServerImpl.java:844)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[grpc-default-executor-4] WARN org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Logging client failed unexpectedly.
org.apache.beam.vendor.grpc.v1p36p0.io.grpc.StatusRuntimeException: CANCELLED: client cancelled
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.Status.asRuntimeException(Status.java:526)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.stub.ServerCalls$StreamingServerCallHandler$StreamingServerCallListener.onCancel(ServerCalls.java:284)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.PartialForwardingServerCallListener.onCancel(PartialForwardingServerCallListener.java:40)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.ForwardingServerCallListener.onCancel(ForwardingServerCallListener.java:23)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onCancel(ForwardingServerCallListener.java:40)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.Contexts$ContextualizedServerCallListener.onCancel(Contexts.java:96)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closedInternal(ServerCallImpl.java:353)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closed(ServerCallImpl.java:341)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1Closed.runInContext(ServerImpl.java:844)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[grpc-default-executor-0] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Beam Fn Logging client connected.
[grpc-default-executor-2] WARN org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Logging client failed unexpectedly.
org.apache.beam.vendor.grpc.v1p36p0.io.grpc.StatusRuntimeException: CANCELLED: client cancelled
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.Status.asRuntimeException(Status.java:526)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.stub.ServerCalls$StreamingServerCallHandler$StreamingServerCallListener.onCancel(ServerCalls.java:284)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.PartialForwardingServerCallListener.onCancel(PartialForwardingServerCallListener.java:40)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.ForwardingServerCallListener.onCancel(ForwardingServerCallListener.java:23)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onCancel(ForwardingServerCallListener.java:40)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.Contexts$ContextualizedServerCallListener.onCancel(Contexts.java:96)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closedInternal(ServerCallImpl.java:353)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closed(ServerCallImpl.java:341)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1Closed.runInContext(ServerImpl.java:844)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at org.apache.beam.vendor.grpc.v1p36p0.io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[grpc-default-executor-5] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Beam Fn Logging client connected.
[grpc-default-executor-6] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Beam Fn Logging client connected.
[grpc-default-executor-5] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Logging client hanged up.
[grpc-default-executor-7] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Beam Fn Logging client connected.
[grpc-default-executor-7] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Logging client hanged up.
[grpc-default-executor-6] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Logging client hanged up.
[grpc-default-executor-6] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Beam Fn Logging client connected.
[grpc-default-executor-9] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Beam Fn Logging client connected.
[grpc-default-executor-8] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - Beam Fn Logging client connected.
[Test worker] INFO org.apache.beam.runners.fnexecution.logging.GrpcLoggingService - 3 Beam Fn Logging clients still connected during shutdown.
```